tests/ProductBundle/Controller/Api/ProductControllerTest.php

```diff
@@ -143,7 +144,7 @@ public function testPostProductAction(): void
         $productPool = $this->createMock(ProductPool::class);
         $productPool->expects($this->once())->method('getManager')->will($this->returnValue($productManager));
 
-        $formatterPool = $this->createMock(FormatterPool::class);
+        $formatterPool = $this->getFormatterPool();
         $formatterPool->expects($this->exactly(2))->method('transform');
```

Comment:
`$formatterPool` isn't a mock instance now.

Reply:
I'm not quite familiar with unit tests. AFAIK this mock is used to determine if the transform method is called twice. But since this shouldn't be a mock anymore, is this check still necessary/possible?

Reply:
The test checking if the method is called is not required anymore since you're using the real object now, so the call to `transform()` (or any other method) will be really performed and the logic from that class will be responsible to produce a valid object.
The goal of the count on the mock was to ensure the involved methods of that class were called, because all the class methods were mocked too.

Reply:
Ok, thanks!